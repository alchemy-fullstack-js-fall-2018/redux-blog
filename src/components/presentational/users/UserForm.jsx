import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Input from '../../lib/Input.jsx';

const StyledDiv = styled.div`
  display: grid;
  grid-template-columns: 300px 300px;
  margin: 0px 15%;
`;

export default class UserForm extends PureComponent {
  static propTypes = {
    registerUser: PropTypes.func.isRequired
  };

  state = {
    name: '',
    username: '',
    email: '',
    phone: '',
    website: '',
    street: '',
    suite: '',
    city: '',
    zipcode: '',
    lat: '',
    lng: '',
    companyName: '',
    motto: '',
    keywords: ''
  };

  onChange = event => {
    const { target } = event;
    console.log(target);
    this.setState({ [target.name]: target.value });
  };

  saveUser = event => {
    event.preventDefault();
    const { name, username, email, phone, website,
      street, suite, city, zipcode,
      companyName, motto, keywords
    } = this.state;
    this.props.registerUser({
      name, username, email, phone, website,
      address: {
        street, suite, city, zipcode
      },
      company: {
        name: companyName, catchPhrase: motto, bs: keywords
      }
    });
    this.props.history.push('/users');
  };

  render() {
    const {
      name, username, email, phone, website,
      street, suite, city, zipcode, lat, lng,
      companyName, motto, keywords
    } = this.state;

    return (
      <form onSubmit={this.saveUser}>

        <h3>Core Info</h3>
        <StyledDiv>
          <Input resource={name} name={'name'} onChange={this.onChange} />
          <Input resource={username} name={'username'} onChange={this.onChange} />
          <Input resource={email} name={'email'} onChange={this.onChange} />
          <Input resource={phone} name={'phone'} onChange={this.onChange} />
          <Input resource={website} name={'website'} onChange={this.onChange} />
        </StyledDiv>
        <h3>Address</h3>
        <StyledDiv>
          <Input resource={street} name={'street'} onChange={this.onChange} />
          <Input resource={suite} name={'suite'} onChange={this.onChange} />
          <Input resource={city} name={'city'} onChange={this.onChange} />
          <Input resource={zipcode} name={'zipcode'} onChange={this.onChange} />
        </StyledDiv>
        <h3>Company Info</h3>
        <StyledDiv>
          <Input resource={companyName} name={'companyName'} onChange={this.onChange} />
          <Input resource={motto} name={'motto'} onChange={this.onChange} />
          <Input resource={keywords} name={'keywords'} onChange={this.onChange} />
        </StyledDiv>

        <button type="submit">Register</button>

      </form>
    );
  }



}
