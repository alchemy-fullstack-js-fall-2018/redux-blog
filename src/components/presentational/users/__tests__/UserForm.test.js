import React from 'react';
import { shallow } from 'enzyme';
import UserForm from '../UserForm.jsx';

describe('<UserForm />', () => {
  test('renders', () => {
    const registerUser = jest.fn();
    const wrapper = shallow(
      <UserForm
        registerUser={registerUser}
        name ={'asdf'}
        username = {'fds'}
        email = {'asfasdf'}
        phone = {'asdasdff'}
        website = {'fsdf'}
        street = {'sdfsdf'}
        suite = {'sdfsdfsdfsdf'}
        city = {'asdfff'}
        zipcode = {'aasssdd'}
        lat = {'fffds'}
        lng = {'asdfa'}
        companyName = {'fasdfasdfafds'}
        motto = {'ffda'}
        keywords = {'asdddf'}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
