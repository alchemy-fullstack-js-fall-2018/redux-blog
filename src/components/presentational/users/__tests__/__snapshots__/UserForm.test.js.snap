// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserForm /> renders 1`] = `
<form
  onSubmit={[Function]}
>
  <h3>
    Core Info
  </h3>
  <styled.div>
    <Input
      name="name"
      onChange={[Function]}
      resource=""
    />
    <Input
      name="username"
      onChange={[Function]}
      resource=""
    />
    <Input
      name="email"
      onChange={[Function]}
      resource=""
    />
    <Input
      name="phone"
      onChange={[Function]}
      resource=""
    />
    <Input
      name="website"
      onChange={[Function]}
      resource=""
    />
  </styled.div>
  <h3>
    Address
  </h3>
  <styled.div>
    <Input
      name="street"
      onChange={[Function]}
      resource=""
    />
    <Input
      name="suite"
      onChange={[Function]}
      resource=""
    />
    <Input
      name="city"
      onChange={[Function]}
      resource=""
    />
    <Input
      name="zipcode"
      onChange={[Function]}
      resource=""
    />
  </styled.div>
  <h3>
    Company Info
  </h3>
  <styled.div>
    <Input
      name="companyName"
      onChange={[Function]}
      resource=""
    />
    <Input
      name="motto"
      onChange={[Function]}
      resource=""
    />
    <Input
      name="keywords"
      onChange={[Function]}
      resource=""
    />
  </styled.div>
  <button
    type="submit"
  >
    Register
  </button>
</form>
`;
